src/main/java/org/mskcc/cbio/oncokb/config/cache/CustomCacheManager.java

```diff
@@ -0,0 +1,46 @@
+package org.mskcc.cbio.oncokb.config.cache;
+
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.concurrent.ConcurrentMapCache;
+
+/**
+ * This cache manager is used when Redis caching is disabled.
+ */
+public class CustomCacheManager implements CacheManager{
+
+    private final ConcurrentMap<String, Cache> cacheMap = new ConcurrentHashMap<>();
+
+    /**
+	 * Get a collection of the cache names known by this manager.
+	 * @return the names of all caches known by the cache manager
+	 */
+    @Override
+	public Collection<String> getCacheNames(){
+        return cacheMap.keySet();
+    }
+
+
+    /**
+	 * Get the cache associated with the given name.
+	 * <p>Note that the cache may be lazily created at runtime if the
+	 * native provider supports it.
+	 * @param name the cache identifier (must not be {@code null})
+	 * @return the associated cache, or {@code null} if such a cache
+	 * does not exist or could be not created
+	 */
+    @Override
+    public Cache getCache(String name) {
+        Cache cache = this.cacheMap.get(name);
+        if (cache == null){
+            cache = new ConcurrentMapCache(name, new ConcurrentHashMap<>(), true);
+            cacheMap.put(name, cache);
+        }
+        return cache;
```

Comment:
Oh I just realize that we actually store the content when the redis is disabled. This memory cache will continue building up?

Reply:
Good point, the cache will buildup and we might run out of memory. I can evict all entries from the cache if cache != null before returning the cache.